src/cargo/util/mod.rs

```diff
@@ -133,6 +134,77 @@ pub fn truncate_with_ellipsis(s: &str, max_width: usize) -> String {
     prefix
 }
 
+#[cfg(not(windows))]
+#[inline]
+pub fn try_canonicalize<P: AsRef<Path>>(path: P) -> std::io::Result<PathBuf> {
+    std::fs::canonicalize(&path)
+}
+
+#[cfg(windows)]
+#[inline]
+pub fn try_canonicalize<P: AsRef<Path>>(path: P) -> std::io::Result<PathBuf> {
+    use std::ffi::OsString;
+    use std::io::Error;
+    use std::os::windows::ffi::{OsStrExt, OsStringExt};
+    use std::{io::ErrorKind, ptr};
+    use windows_sys::Win32::Foundation::{GetLastError, SetLastError};
+    use windows_sys::Win32::Storage::FileSystem::GetFullPathNameW;
+
+    // On Windows `canonicalize` may fail, so we fall back to getting an absolute path.
+    std::fs::canonicalize(&path).or_else(|_| {
```

Comment:
Should we wait for the stabilization of `std::path::aboslute`? We should also have a TODO here and call out where this piece of code come from. (Looks like [`fn absolute`](https://github.com/rust-lang/rust/blob/da7c50c089d5db2d3ebaf227fe075bb1346bfaec/library/std/src/sys/windows/path.rs#L308-L333) on windows to me.)

Reply:
That would probably take a while.